### PR TITLE
[IMP] misc: email add new alias sys parameter

### PR DESCRIPTION
**Pull request by tiku-odoo**
Task: https://www.odoo.com/web#id=3667301&cids=3&menu_id=4720&action=333&active_id=3835&model=project.task&view_type=form

### CODE REVIEW

**Comment by tiku-odoo** [comment]
@AutumnWalk 

When you have a moment, can you review lines: 275-306?

Thanks, 
Tim :+1:

**Comment by tiku-odoo** [comment]
@jero-odoo 

When you have a moment, can you review lines: 275-306?

The rest of the doc has been reviewed in the last 6-12 months. 

Thanks in advance :+1: 

 Tim

**Comment by tiku-odoo** [comment]
@jero-odoo 

Bumping this to the top of your inbox. Can you review lines: 275-306?

The rest of the doc has been reviewed in the last 6-12 months.

Thanks, Tim

File: content/applications/general/email_communication/email_servers.rst [hunk 1]
```diff
@@ -21,7 +21,7 @@ standard online Odoo database normally since it has already been pre-configured
 Scope of this documentation
 ===========================
 
-This document is **mainly dedicated to Odoo on-premise databases** who don't benefit from an
+This document is **mainly dedicated to Odoo on-premise databases** who do not benefit from an
```
Comment (jero-odoo) [inline content/applications/general/email_communication/email_servers.rst:24]:
```suggestion
This document is **mainly dedicated to Odoo on-premise databases** that do not benefit from an
```

File: content/applications/general/email_communication/email_servers.rst [hunk 2]
```diff
@@ -263,11 +263,44 @@ To edit catchall and bounce aliases, first activate the :ref:`developer mode <de
 Then, go to :menuselection:`Settings --> Technical --> Parameters --> System Parameters` to
 customize the aliases (`mail.catchall.alias` & `mail.bounce.alias`). These types of changes should
 be completed prior to the database going live. If a customer replies after a change is made then the
-system will not recognize the old alias and the reply won't be received.
+system will not recognize the old alias and the reply will not be received.
```
Comment (jero-odoo) [inline content/applications/general/email_communication/email_servers.rst:266]:
```suggestion
system will not recognize the old alias, and the reply will not be received.
```

File: content/applications/general/email_communication/email_servers.rst [hunk 3]
```diff
@@ -263,11 +263,44 @@ To edit catchall and bounce aliases, first activate the :ref:`developer mode <de
...
+-----------------------------------
+
+Incoming aliases are set in the Odoo database to create records by receiving incoming emails. To
+view aliases set in the Odoo database, first activate the :ref:`developer mode <developer-mode>`.
+Then, go to :menuselection:`Settings --> Technical --> Email --> Aliases`.
+
+The following system parameter, `mail.catchall.domain.allowed`, set with allowed alias domain
+values, separated by commas, will filter out correctly addressed emails to aliases. By setting the
+domain(s) for which the alias is allowed to create a ticket/lead opportunity/etc, this eliminates
+false positives where email addresses with only the prefix alias (and not the domain) is present.
```
Comment (jero-odoo) [inline content/applications/general/email_communication/email_servers.rst:285]:
```suggestion
values, separated by commas, will filter out correctly addressed emails to aliases. Setting the
domain(s) for which the alias can create a ticket, lead, opportunity, etc., eliminates false positives where email addresses with only the prefix alias (not the domain) are present.
```

File: content/applications/general/email_communication/email_servers.rst [hunk 4]
```diff
@@ -263,11 +263,44 @@ To edit catchall and bounce aliases, first activate the :ref:`developer mode <de
...
+
+In some instances matches have been made in the Odoo database when an email is received with the
```
Comment (jero-odoo) [inline content/applications/general/email_communication/email_servers.rst:287]:
```suggestion
In some instances, matches have been made in the Odoo database when an email is received with the
```

File: content/applications/general/email_communication/email_servers.rst [hunk 5]
```diff
@@ -263,11 +263,44 @@ To edit catchall and bounce aliases, first activate the :ref:`developer mode <de
...
+   For example, when Odoo receives emails that have the name `commercial` prefix alias in the
+   sender, recipient, or :abbr:`CC (Carbon Copy)` email address(es) (commercial@gmail.com,
+   commercial@odoo.net), the database will falsely treat the email as the full `commercial` alias
+   (with a different domain) and therefore create a ticket/lead/opportunity/etc..
+
+To add the `mail.catchall.domain.allowed` system parameter, first activate the :ref:`developer mode
+<developer-mode>`. Then, go to :menuselection:`Settings --> Technical --> Parameters --> System
+Parameters`. Click, :guilabel:`Create` / :guilabel:`New`. Then type in
+`mail.catchall.domain.allowed` for the :guilabel:`Key` field. Next, for the :guilabel:`Value` field,
+add the domain(s) separated by comma(s) (if plural domains). Manually :guilabel:`Save` and the
```
Comment (jero-odoo) [inline content/applications/general/email_communication/email_servers.rst:301]:
```suggestion
add the domain(s) separated by comma(s) (if plural domains). Manually :guilabel:`Save`, and the
```

File: content/applications/general/email_communication/email_servers.rst [hunk 6]
```diff
@@ -263,11 +263,44 @@ To edit catchall and bounce aliases, first activate the :ref:`developer mode <de
...
+same alias prefix and a different domain on the incoming email address. This is true in the sender,
+recipient, and :abbr:`CC (Carbon Copy)` email addresses of an incoming email.
+
+.. example::
+   For example, when Odoo receives emails that have the name `commercial` prefix alias in the
+   sender, recipient, or :abbr:`CC (Carbon Copy)` email address(es) (commercial@gmail.com,
+   commercial@odoo.net), the database will falsely treat the email as the full `commercial` alias
+   (with a different domain) and therefore create a ticket/lead/opportunity/etc..
+
+To add the `mail.catchall.domain.allowed` system parameter, first activate the :ref:`developer mode
```
Comment (jero-odoo) [inline content/applications/general/email_communication/email_servers.rst:297]:
```suggestion
To add the `mail.catchall.domain.allowed` system parameter, first, activate the :ref:`developer mode
```

**Review by jero-odoo — APPROVED** [review]
@tiku-odoo just a few formatting suggestions, otherwise this looks good to me 👍

**Comment by tiku-odoo** [comment]
@ksc-odoo 

This doc is ready for your review. 

Can you review lines: 275-306?

The rest of the doc has been reviewed in the last 6-12 months.

Thanks,  Tim

File: content/applications/general/email_communication/email_servers.rst [hunk 7]
```diff
@@ -263,11 +263,44 @@ To edit catchall and bounce aliases, first activate the :ref:`developer mode <de
...
+
+Allow alias domain system parameter
+-----------------------------------
+
+Incoming aliases are set in the Odoo database to create records by receiving incoming emails. To
+view aliases set in the Odoo database, first activate the :ref:`developer mode <developer-mode>`.
+Then, go to :menuselection:`Settings --> Technical --> Email --> Aliases`.
+
+The following system parameter, `mail.catchall.domain.allowed`, set with allowed alias domain
+values, separated by commas, will filter out correctly addressed emails to aliases. Setting the
```
Comment (ksc-odoo) [inline content/applications/general/email_communication/email_servers.rst:283]:
```suggestion
values, separated by commas, filters out correctly addressed emails to aliases. Setting the
```

File: content/applications/general/email_communication/email_servers.rst [hunk 8]
```diff
@@ -263,11 +263,44 @@ To edit catchall and bounce aliases, first activate the :ref:`developer mode <de
...
+
+In some instances, matches have been made in the Odoo database when an email is received with the
+same alias prefix and a different domain on the incoming email address. This is true in the sender,
+recipient, and :abbr:`CC (Carbon Copy)` email addresses of an incoming email.
+
+.. example::
+   For example, when Odoo receives emails that have the name `commercial` prefix alias in the
+   sender, recipient, or :abbr:`CC (Carbon Copy)` email address(es) (commercial@gmail.com,
+   commercial@odoo.net), the database will falsely treat the email as the full `commercial` alias
+   (with a different domain) and therefore create a ticket/lead/opportunity/etc..
```
Comment (ksc-odoo) [inline content/applications/general/email_communication/email_servers.rst:295]:
```suggestion
   When Odoo receives emails that have the name `commercial` prefix alias in the sender, recipient,
   or :abbr:`CC (Carbon Copy)` email address(es) (e.g. commercial@gmail.com, commercial@odoo.net),
   the database falsely treats the email as the full `commercial` alias (with a different domain),
   and therefore, creates a ticket/lead/opportunity/etc..
```

File: content/applications/general/email_communication/email_servers.rst [hunk 9]
```diff
@@ -263,11 +263,44 @@ To edit catchall and bounce aliases, first activate the :ref:`developer mode <de
...
+
+To add the `mail.catchall.domain.allowed` system parameter, first, activate the :ref:`developer mode
+<developer-mode>`. Then, go to :menuselection:`Settings --> Technical --> Parameters --> System
+Parameters`. Click, :guilabel:`Create` / :guilabel:`New`. Then type in
+`mail.catchall.domain.allowed` for the :guilabel:`Key` field. Next, for the :guilabel:`Value` field,
+add the domain(s) separated by comma(s) (if plural domains). Manually :guilabel:`Save`, and the
+system parameter takes immediate effect.
```
Comment (ksc-odoo) [inline content/applications/general/email_communication/email_servers.rst:302]:
```suggestion
<developer-mode>`. Then, go to :menuselection:`Settings app --> Technical --> Parameters section -->
System Parameters`. Click :guilabel:`Create`. Then, type in `mail.catchall.domain.allowed` for the
:guilabel:`Key` field.

Next, for the :guilabel:`Value` field, add the domain(s) separated by comma(s) (if plural domains).
Manually :guilabel:`Save`, and the system parameter takes immediate effect.
```

**Review by ksc-odoo — APPROVED** [review]
@tiku-odoo Great work on this PR. I just conducted a Final Review for the requested lines, and I'm approving because there is only a few small adjustments that require your attention. Once you make those necessary changes, you can feel free to move this one to the next stage. Thanks! 👍

**Comment by tiku-odoo** [comment]
@samueljlieber 

This doc is ready for your review when you have a moment. Thanks :+1:

File: content/applications/general/email_communication/email_servers.rst [hunk 10]
```diff
@@ -263,11 +263,45 @@ To edit catchall and bounce aliases, first activate the :ref:`developer mode <de
...
+
+In some instances, matches have been made in the Odoo database when an email is received with the
+same alias prefix and a different domain on the incoming email address. This is true in the sender,
+recipient, and :abbr:`CC (Carbon Copy)` email addresses of an incoming email.
+
+.. example::
+   When Odoo receives emails that have the name `commercial` prefix alias in the sender, recipient,
+   or :abbr:`CC (Carbon Copy)` email address(es) (e.g. commercial@gmail.com, commercial@odoo.net),
+   the database falsely treats the email as the full `commercial` alias (with a different domain),
+   and therefore, creates a ticket/lead/opportunity/etc..
```
Comment (samueljlieber) [inline content/applications/general/email_communication/email_servers.rst:295]: Lets add literal formatting to these email addresses to be consistent with the rest of this documentation :)
```suggestion
   or :abbr:`CC (Carbon Copy)` email address(es) (e.g. `commercial\@gmail.com`,
   `commercial\@odoo.net`), the database falsely treats the email as the full `commercial` alias
   (with a different domain), and therefore, creates a ticket/lead/opportunity/etc..
```

**Review by samueljlieber — APPROVED** [review]
Hi @tiku-odoo great job with the improvements in this PR, everything looks great. I have one small suggestion, please see below. I am approving now, feel free to tag ZST for merge if you choose to implement this suggestion :)

File: content/applications/general/email_communication/email_servers.rst (hunk 10 above)
Comment (StraubCreative) [reply content/applications/general/email_communication/email_servers.rst:295]: Agree with Sam's comment.

Also on line 295: either add another period to form a complete ellipses OR remove the second period to end the sentence

```suggestion
   (with a different domain), and therefore, creates a ticket/lead/opportunity/etc...
```

**Review by StraubCreative — APPROVED** [review]
Approving with comment
Merge when ready @tiku-odoo 👍 
..
@robodoo delegate=tiku-odoo

**Comment by tiku-odoo** [comment]
@robodoo r+

**Pull request closed without merging**